hadoop-ozone/common/src/main/java/org/apache/hadoop/ozone/om/protocolPB/OMAdminProtocolClientSideImpl.java

```diff
@@ -123,12 +175,48 @@ public OMConfiguration getOMConfiguration() throws IOException {
       }
       return omMedatataBuilder.build();
     } catch (ServiceException e) {
-      LOG.error("Failed to retrieve configuration of OM {}",
-          remoteOmNodeDetails.getOMPrintInfo(), e);
+      LOG.error("Failed to retrieve configuration of OM {}", omPrintInfo, e);
     }
     return null;
   }
 
+  @Override
+  public void decommission(OMNodeDetails removeOMNode) throws IOException {
+    DecommissionOMRequest decommOMRequest = DecommissionOMRequest.newBuilder()
+        .setNodeId(removeOMNode.getNodeId())
+        .setNodeAddress(removeOMNode.getHostAddress())
+        .build();
+
+    DecommissionOMResponse response;
+    try {
+      response = rpcProxy.decommission(NULL_RPC_CONTROLLER, decommOMRequest);
```

Comment:
When reconfiguration exception, when already a in progress request, in failOver we need to retry on same OM for the ReconfigurationInProgressException

Reply:
Good catch Bharat.. I can update the OMFailoverProxyProvider#getRetryPolicy() to include this condition for ReconfigurationInProgressException. But this condition need not be checked for regular client requests.

So I was thinking we should improve the retry policy to be specific for each protocol instead of a generic one. There is a Jira (HDDS-5847) open for adding a separate failover proxy for OMInterServiceProtocol (ReconfigurationInProgressException is valid for inter service protocol too). I can update this along with that, if you are ok with that?

Reply:
Ya sure. Pls update HDDS-5847 so that we don't miss it.